CMakeLists.txt

```diff
@@ -6,6 +6,8 @@ endif()
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
```

Comment:
It slipped in by accident. It makes CMake generate a file, compile_commsnds.json, that contains all the flags used by the different parts of the build. The auto-completion tool I use needs this file to provide semantic completion suggestions. I'd love to have it in master, but that should be its own pull request. I'll remove it.